.github/workflows/docs-python.yml

```diff
@@ -51,7 +51,7 @@ jobs:
         run: make html
 
       - name: Deploy Python docs for latest development version
-        if: ${{ github.ref_name == 'main' }}
+        if: github.event_name == 'push' && github.ref_name == 'main'
```

Comment:
Drive-by: we don't need to build the dev docs on Python release, as they are already updated with every push to main.